java/src/jmri/jmrit/dispatcher/TrainInfoFile.java

```diff
@@ -532,6 +535,7 @@ public void writeTrainInfo(TrainInfo tf, String name) throws java.io.IOException
         traininfo.setAttribute("usespeedprofile", "" + (tf.getUseSpeedProfile() ? "yes" : "no"));
         traininfo.setAttribute("stopbyspeedprofile", "" + (tf.getStopBySpeedProfile() ? "yes" : "no"));
         traininfo.setAttribute("stopbyspeedprofileadjust", Float.toString(tf.getStopBySpeedProfileAdjust()));
+        traininfo.setAttribute("waittime", Float.toString(tf.getWaitTime()));
```

Comment:
The file `xml/DTD/dispatcher-traininfo.dtd` needs to be updated.

Reply:
I saw a comment that xml/DTD/dispatcher-traininfo.dtd had to be updated, but I could not find it. I presume I am looking in the wrong directory

![image](https://user-images.githubusercontent.com/8853745/196246033-5567660d-cf33-4ab7-a300-4d7a2a292cd7.png)

Reply:
Expand the xml line.

Reply:
Searched for it, and intellij found it, but I don't know why it is not showing it in the file structure...

Reply:
It turns out that there are a couple of missing attributes.
```
     <!ATTLIST traininfo trainfromuser          CDATA #REQUIRED>
+    <!ATTLIST traininfo trainfromsetlater              CDATA #REQUIRED>
     <!ATTLIST traininfo priority               CDATA #REQUIRED>
     <!ATTLIST traininfo resetwhendone          CDATA #REQUIRED>
     <!ATTLIST traininfo delayedrestart         CDATA "no">
@@ -56,8 +57,10 @@
     <!ATTLIST traininfo terminatewhendone      CDATA "yes">
     <!ATTLIST traininfo loadatstartup          CDATA "no">
     <!ATTLIST traininfo allocationmethod       CDATA "3">
+    <!ATTLIST traininfo nexttrain              CDATA #IMPLIED>
     <!ATTLIST traininfo allocatealltheway      CDATA "no">
```

Reply:
would be nice to know why the system worked with errors in the file. Is the file really necessary?

Reply:
The process for loading train info files does not currently validate the content.  If that changes in the future then the DTD needs to be correct.

You can manually validate any xml file using the **Debug &rArr; Validate XML File** menu item.